src/qibolab/instruments/rfsoc.py

```diff
@@ -182,14 +278,14 @@ def _open_connection(host: str, port: int, server_commands: dict):
                 if not tmp:
                     break
                 received.extend(tmp)
-        results = pickle.loads(received)
-        if isinstance(results, Exception):
-            raise results
+        results = json.loads(received)
```

Comment:
Here there is a mild asymmetry: during transmission you're encoding your string in bytes, with a specific encoding, instead while receiving you're using it implicitly.

Is the asymmetry generated by the standard library or only here?

Reply:
To the server I can send only bytes so I need to do `bytes(json.dump(...)))`, on the other hand `json.loads` can decode it automatically